### PR TITLE
Added Landing Page

### PR DESCRIPTION
**Pull request by LukasLJL**
I have added/copied a landing page like Synpase does.
Recently I have installed Dendrite and was wondering why it´s not working. After some troubleshooting I figured out there is no landing page like synpase has, so the Server was running just fine. 

Hopefuly this PR can fix this problem and may help other users who run into this issue.
I have not written any unit tests, because it´s just a simple landing page with a redirect to a static site.

### Pull Request Checklist

* [x] I have added Go unit tests or [Complement integration tests](https://github.com/matrix-org/complement) for this PR _or_ I have justified why this PR doesn't need tests
* [x] Pull request includes a [sign off below using a legally identifiable name](https://matrix-org.github.io/dendrite/development/contributing#sign-off) _or_ I have already signed off privately

Signed-off-by: `Lukas Huida<lukas@leucali.net>`

### CODE REVIEW

**Comment by kegsay** [comment]
This needs tests to ensure the routing is correct.

**Review by kegsay — CHANGES_REQUESTED** [review]
Needs tests to make sure routing is working correctly in monolith mode at the very least. Otherwise LGTM!

**Comment by LukasLJL** [comment]
Thanks for the reply, could you give me a hint where I can add the test? Are there any example?

**Comment by kegsay** [comment]
There aren't as there hasn't been anything like this before. You need a `setup.Monolith` and a `base.Dendrite` and to glue them together similar to https://github.com/matrix-org/dendrite/blob/main/cmd/dendrite-monolith-server/main.go - then you can make a fake request with [httptest](https://pkg.go.dev/net/http/httptest#NewRequest) and pass it to the handler to ensure you get back the HTML page.

**Comment by S7evinK** [comment]
We should probably also embed the html file and/or make it a template to include the actual Dendrite version.
Something like
```html
    <h1>It works! Dendrite {{ .dendriteVersion }} is running</h1>
```
would be great.

File: setup/base/base.go [hunk 1]
```diff
@@ -478,13 +486,30 @@ func (b *BaseDendrite) SetupAndServeHTTP(
 
 	b.configureHTTPErrors()
 
+	//Redirect for Landing Page
+	externalRouter.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, httputil.PublicStaticPath, http.StatusSeeOther)
```
Comment (kegsay) [inline setup/base/base.go:491]: `http.StatusFound` surely?

File: setup/base/base.go [hunk 2]
```diff
@@ -478,13 +486,30 @@ func (b *BaseDendrite) SetupAndServeHTTP(
...
 		internalRouter.Handle("/metrics", httputil.WrapHandlerInBasicAuth(promhttp.Handler(), b.Cfg.Global.Metrics.BasicAuth))
 	}
 
 	b.ConfigureAdminEndpoints()
 
+	tmpl := template.Must(template.ParseFS(staticContent, "static/*.gotmpl"))
+
+	b.PublicStaticMux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		if err := tmpl.ExecuteTemplate(w, "index.gotmpl", map[string]string{
+			"Version": internal.VersionString(),
```
Comment (kegsay) [inline setup/base/base.go:505]: It seems inefficient to template this every time the client asks. Just do it once at startup and return those bytes?

**Review by kegsay — APPROVED** [review]
Otherwise LGTM

**Pull request merged**